web3/contracts/MirrorMarketplace.sol

```diff
@@ -0,0 +1,25 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract MirrorMarketplace {
+    uint listingId;
+
+    enum ListingStatus {
+        Listed,
+        Cancelled,
+        Sold
+    }
+
+    struct Listing {
+        IERC721 token;
+        address seller;
+        uint tokenId;
+        uint price;
```

Comment:
so I was going to say shouldn't be this a floating number, but seems like there's no floating numbers in solidity? :D

Reply:
yes there is no floating numbers in solidity. but we set a price in gwei so i do not think we need a floating number

Reply:
ok makes sene !